src/Microsoft.IdentityModel.Tokens.Saml/Saml2/Saml2SecurityTokenHandler.cs

```diff
@@ -644,8 +644,12 @@ protected virtual Saml2Conditions CreateConditions(SecurityTokenDescriptor token
             else if (SetDefaultTimesOnTokenCreation)
                 conditions.NotOnOrAfter = DateTime.UtcNow + TimeSpan.FromMinutes(TokenLifetimeInMinutes);
 
-            if (!string.IsNullOrEmpty(tokenDescriptor.Audience))
-                conditions.AudienceRestrictions.Add(new Saml2AudienceRestriction(tokenDescriptor.Audience));
+            var audienceRestriction = new Saml2AudienceRestriction(tokenDescriptor.Audiences.Where(aud => !string.IsNullOrWhiteSpace(aud)));
```

Comment:
Do not use the Where logic, this creates a new enumeration.
Let the Saml2Serializer handle nulls.
Write tests for null values to see what happens.

Reply:
Why is the logic in creating the Uri list different between Saml and Saml2?
Could that not be shared logic?

Reply:
I removed the Where clause.

As for why it is in two places, the short answer is because that is how it was already implemented and refactoring existing public APIs is not in scope except where necessary.

Long answer:
In Saml you need to make a SamlConditions object and add a SamlAudienceRestrictionCondition to it which takes IEnumerable<Uri> as an arg while in Saml2 you need to make a Saml2Conditions object and give it a Saml2AudienceRestriction object which takes an IEnumerable<string> as an arg and thus does not require changing the strings into Uri objects. It would need to be refactored if we wanted consistency in implementation.